modules/apps/object/object-web/src/main/resources/META-INF/resources/js/tests/ModelBuilder/utils.spec.ts

```diff
@@ -209,3 +219,101 @@ describe('checkPostalAddressUnsupportedObjectRelationship function', () => {
 		expect(unsupportedObjectRelationship).toBe(false);
 	});
 });
+
+describe('getUnsupportedObjectRelationshipErrorMessage function', () => {
+	it('returns errorMessage when source and target object definition nodes are unmodifiable', () => {
+		const unmodifiableObjectDefinition = buildObjectDefinitionNodeMock({
+			modifiable: false,
+			objectDefinitionExternalReferenceCode: 'customObjectDefinitionERC',
+			objectDefinitionName: 'customObject',
+		});
+
+		const unmodifiableObjectDefinition2 = buildObjectDefinitionNodeMock({
+			modifiable: false,
+			objectDefinitionExternalReferenceCode: 'customObjectDefinitionERC2',
+			objectDefinitionName: 'customObject2',
+		});
+
+		const errorMessage = getUnsupportedObjectRelationshipErrorMessage(
+			[unmodifiableObjectDefinition, unmodifiableObjectDefinition2],
+			unmodifiableObjectDefinition,
+			unmodifiableObjectDefinition2
+		);
+
+		expect(errorMessage).toMatchObject({
+			errorMessage: 'unmodifiable-system-objects-cannot-be-related',
+		});
+	});
+
+	it('returns errorMessage when the source node is a linked object definition', () => {
+		const linkedObjectDefinition = buildObjectDefinitionNodeMock({
+			linkedObjectDefinition: true,
+			objectDefinitionExternalReferenceCode: 'customObjectDefinitionERC',
+			objectDefinitionName: 'customObject',
+		});
+
+		const customObjectDefinition2 = buildObjectDefinitionNodeMock({
+			objectDefinitionExternalReferenceCode: 'customObjectDefinitionERC2',
+			objectDefinitionName: 'customObject2',
+		});
+
+		const errorMessage = getUnsupportedObjectRelationshipErrorMessage(
+			[linkedObjectDefinition, customObjectDefinition2],
+			linkedObjectDefinition,
+			customObjectDefinition2
+		);
+
+		expect(errorMessage).toMatchObject({
+			errorMessage:
+				'adding-relationships-directly-to-linked-objects-is-not-allowed',
+		});
+	});
+
+	it('returns errorMessage when source and target object definition nodes have Salesforce storageType', () => {
+		const salesforceObjectDefinition = buildObjectDefinitionNodeMock({
+			objectDefinitionExternalReferenceCode: 'customObjectDefinitionERC',
+			objectDefinitionName: 'customObject',
+			storageType: 'salesforce',
+		});
+
+		const salesforceObjectDefinition2 = buildObjectDefinitionNodeMock({
+			objectDefinitionExternalReferenceCode: 'customObjectDefinitionERC2',
+			objectDefinitionName: 'customObject2',
+			storageType: 'salesforce',
+		});
+
+		const errorMessage = getUnsupportedObjectRelationshipErrorMessage(
+			[salesforceObjectDefinition, salesforceObjectDefinition2],
+			salesforceObjectDefinition,
+			salesforceObjectDefinition2
+		);
+
+		expect(errorMessage).toMatchObject({
+			errorMessage: 'salesforce-objects-do-not-support-relationships',
+		});
+	});
+
+	it('returns errorMessage when an unsupported postal address object relationship is checked', () => {
```

Comment:
instead of  '`returns errorMessage when an unsupported postal address object relationship is checked'` you could say `'returns errorMessage when there is an unsupported postal address object relationship'`, what do you think?

Reply:
Thank you for the suggestion @rebsilva!